package.tf

```
@@ -26,10 +26,13 @@ data "external" "archive_prepare" {
       docker_entrypoint         = var.docker_entrypoint
     }) : null
 
-    artifacts_dir = var.artifacts_dir
-    runtime       = var.runtime
-    source_path   = jsonencode(var.source_path)
-    hash_extra    = var.hash_extra
+    artifacts_dir            = var.artifacts_dir
+    runtime                  = var.runtime
+    compatible_runtimes      = var.compatible_runtimes
+    architectures            = jsonencode(var.architectures)
```

```suggestion
    architectures            = var.architectures
```

Right? I didn't run the code, but I think it should not be `jsonencoded` everywhere except `sourch_path`, whic has a very dynamic structure (by design).

Should this be applied to `compatible_architectures`, too?

---

I didn't run this PR, so I don't know if the proposed change will work. Please try it yourself to make sure that PR works as you expect :)

---

Sorry, for the delayed response, but I wanted to thoroughly look at this PR again before getting back to you and now finally found the time to do so.

Without using `jsonencode()` it doesn't work, as the `external` data source expects attributes of the `query` attribute to be strings (https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/external), while the values provided here are lists of strings. I've fixed that now, by adding the missing `jsonencode()` for `var.compatible_runtimes`.

While testing the fix, I also noticed that deployment packages would still be rebuilt, if the order of items in the list of values for the attributes changed. I tried to avoid that originally as well, but failed to recognize that the `sort_keys` of `json.dumps()` only affects dictionaries and not lists as well. I fixed that as well, so changing the order of items in the list of values for the attributes doesn't cause a rebuild of the deployment package either anymore.

From my perspective this PR is good to go again now.